package/MDAnalysis/analysis/msd.py

```diff
@@ -0,0 +1,335 @@
+# J. Comput. Chem. 32 (2011), 2319--2327, doi:10.1002/jcc.21787
+#
+# This module written by Hugo MacDermott-Opeskin 2020
+
+r"""
+Mean Squared Displacement --- :mod:`MDAnalysis.analysis.msd`
+==============================================================
+
+This module implements the calculation of Mean Squared Displacmements (MSDs) by the Einstein relation.
+MSDs can be used to characterise the speed at which particles move and has its roots
+in the study of Brownian motion. For a full explanation of the theory behind MSDs and the subsequent calculation of self diffusivities the reader is directed to [Maginn2019]_.
+MSDs can be computed from the following expression, known as "Einstein" formula:
+
+.. math::
+
+   MSD(r_{d}) = \bigg{\langle} \frac{1}{N} \sum_{i=1}^{N} |r_{d}  - r_{d}(t_0)|^2 \bigg{\rangle}_{t_{0}}
+
+Where :math:`N` is the number of equivalent particles the MSD is calculated over, :math:`r` are their coordinates and :math:`d` the desired
+dimensionality of the MSD. Note that while the definition of the MSD is universal, there are many practical considerations to computing the MSD
+that vary between implementations. In this module, we compute a "windowed" MSD, where the MSD is averaged over all possible lag times :math:`\tau \le \tau_{max}`,
+where :math:`\tau_{max}` is the length of the trajectory, thereby maximising the number of samples.
+
+The computation of the MSD in this way can be computationally intensive due to it's :math:`N^2` scaling with respect to :math:`\tau_{max}`. 
+An algorithm to compute the MSD with :math:`N log(N)` scaling based on a Fast Fourier Transform is known and can be accessed by setting fft=True [Calandri2011]_.
+
+Computing an MSD
+----------------
+This example computes a 3D MSD for the movement of 100 particles undergoing a random walk.
+Files provided as part of the MDAnalysis test suite are used
+(in the variables :data:`~MDAnalysis.tests.datafiles.RANDOM_WALK` and
+:data:`~MDAnalysis.tests.datafiles.RANDOM_WALK_TOPO`)
+
+First load all modules and test data
+
+    >>> import MDAnalysis as mda
+    >>> import MDAnalysis.analysis.msd as msd
+    >>> from MDAnalysis.tests.datafiles import RANDOM_WALK, RANDOM_WALK_TOPO
+
+Given a universe containing trajectory data we can extract the MSD
+Analyis by using the class :class:`EinsteinMSD` 
+
+    >>> u = mda.Universe(RANDOM_WALK, RANDOM_WALK_TOPO)
+    >>> MSD = msd.EinsteinMSD(u, 'all', msd_type='xyz', fft=True)
+    >>> MSD.run()
+
+The MSD can then be accessed as
+
+    >>> msd =  MSD.timeseries
+
+Visual inspection of the MSD is important, so lets take a look at it with a simple plot.
+
+    >>> import matplotlib.pyplot as plt
+    >>> nframes = MSD.N_frames
+    >>> timestep = 1 # this needs to be the actual time between frames in your trajectory
+    >>> lagtimes = np.arange(nframes)*timestep # make the lag time axis
+    >>> plt.plot(msd, lagtimes)
+    >>> plt.show()
```

Comment:
If you want to include an example image then you can do so and add it under [doc/sphinx/source/images](https://github.com/MDAnalysis/mdanalysis/tree/develop/package/doc/sphinx/source/images); see https://github.com/MDAnalysis/mdanalysis/blob/39b6385b358a7b38d52693b2345620292a2b4297/package/MDAnalysis/analysis/dihedrals.py#L90-L100 for an example how to include it.